test/e2e_node/util.go

```diff
@@ -408,6 +408,9 @@ func restartKubelet(running bool) {
 
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kubeletServiceName).CombinedOutput()
 	framework.ExpectNoError(err, "Failed to restart kubelet with systemctl: %v, %s", err, string(stdout))
+
+	// testing, some test failed after kubelet restarts.
+	time.Sleep(5 * time.Second)
```

Comment:
and a wait loop checking for `systemct status kubelet`?

Reply:
good idea.

Some special cases: However, what if kubelet crashed after it started quickly. The check may pass but kubelet crashes then. (If so, the test should fail.)

```
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/resource_usage_test.go:86
Unexpected error:
    <*errors.errorString | 0xc001746000>: {
        s: "No resource usage data for kubelet container (/kubelet.slice)",
    }
    No resource usage data for kubelet container (/kubelet.slice)
occurred
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/resource_usage_test.go:194
```

Reply:
just maintain the condition for X seconds ?
wait until `systemctl status is ok during 3 seconds`